action.yml

```diff
@@ -0,0 +1,323 @@
+        fi
+        
+        echo "Extracting binaryen..."
+        tar -xzf binaryen.tar.gz
+        rm binaryen.tar.gz
+        
+        # Find the extracted directory and move binaries
+        BINARYEN_DIR=$(find . -maxdepth 1 -type d -name "binaryen-*" | head -n 1)
+        if [ -z "$BINARYEN_DIR" ]; then
+          echo "Failed to find extracted binaryen directory"
+          exit 1
+        fi
+        
+        INSTALL_DIR="$HOME/.wasixcc/bin"
+        mkdir -p "$INSTALL_DIR"
+        
+        # Copy binaries
+        if [ "$OS" = "windows" ]; then
+          if cp "$BINARYEN_DIR/bin/"*.exe "$INSTALL_DIR/" 2>/dev/null; then
+            echo "Copied binaryen binaries successfully"
+          else
+            echo "Warning: Failed to copy some binaryen binaries"
+          fi
+        else
+          if cp "$BINARYEN_DIR/bin/"* "$INSTALL_DIR/" 2>/dev/null; then
+            chmod +x "$INSTALL_DIR/"*
+            echo "Copied binaryen binaries successfully"
+          else
+            echo "Warning: Failed to copy some binaryen binaries"
+          fi
+        fi
+        
+        rm -rf "$BINARYEN_DIR"
+        
+        # Verify wasm-opt was installed
+        if [ -f "$INSTALL_DIR/wasm-opt" ] || [ -f "$INSTALL_DIR/wasm-opt.exe" ]; then
+          echo "Installed binaryen successfully"
+          "$INSTALL_DIR/wasm-opt" --version
+        else
+          echo "Error: wasm-opt not found after installation"
+          exit 1
+        fi
+
+    - name: Install WASIX sysroot
+      if: inputs.install-sysroot == 'true'
+      id: install-sysroot
+      shell: bash
+      run: |
+        echo "Installing WASIX sysroot..."
+        SYSROOT_TAG="${{ inputs.sysroot-tag }}"
+        
+        if [ "$SYSROOT_TAG" = "latest" ]; then
+          "$HOME/.wasixcc/bin/wasixcc" --download-sysroot
+        else
+          "$HOME/.wasixcc/bin/wasixcc" --download-sysroot "$SYSROOT_TAG"
+        fi
+        
+        # Get and export sysroot path
+        SYSROOT_PATH=$("$HOME/.wasixcc/bin/wasixcc" --print-sysroot)
+        echo "path=$SYSROOT_PATH" >> $GITHUB_OUTPUT
+        echo "Installed WASIX sysroot to: $SYSROOT_PATH"
+
+    - name: Install LLVM toolchain
+      if: inputs.install-llvm == 'true'
+      shell: bash
+      run: |
+        echo "Installing LLVM toolchain..."
+        LLVM_TAG="${{ inputs.llvm-tag }}"
+        
+        if [ "$LLVM_TAG" = "latest" ]; then
+          "$HOME/.wasixcc/bin/wasixcc" --download-llvm
+        else
+          "$HOME/.wasixcc/bin/wasixcc" --download-llvm "$LLVM_TAG"
+        fi
+        
+        echo "Installed LLVM toolchain successfully"
+
+    - name: Verify installation
+      shell: bash
+      run: |
+        echo "Verifying installation..."
+        echo "wasixcc version:"
+        wasixcc --version
+        
+        if [ "${{ inputs.install-binaryen }}" = "true" ]; then
+          echo "wasm-opt version:"
+          wasm-opt --version || echo "Warning: wasm-opt not found"
+        fi
```

Comment:
I think wasm-opt isn't directly accessible if it is only installed via wasixcc, so this test will fail